railties/lib/rails/generators/rails/app/templates/Gemfile

```diff
@@ -1,5 +1,6 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+ruby <%= "'#{RUBY_VERSION}'" %>
```

Comment:
I use this "trick", but I haven't tried it with `heroku`, so maybe it doesn't work well, not sure.